src/terminal/parser/IStateMachineEngine.hpp

```diff
@@ -26,33 +26,26 @@ namespace Microsoft::Console::VirtualTerminal
         virtual bool ActionExecute(const wchar_t wch) = 0;
         virtual bool ActionExecuteFromEscape(const wchar_t wch) = 0;
         virtual bool ActionPrint(const wchar_t wch) = 0;
-        virtual bool ActionPrintString(const wchar_t* const rgwch,
-                                       size_t const cch) = 0;
+        virtual bool ActionPrintString(const std::wstring_view string) = 0;
 
-        virtual bool ActionPassThroughString(const wchar_t* const rgwch,
-                                             size_t const cch) = 0;
+        virtual bool ActionPassThroughString(const std::wstring_view string) = 0;
 
         virtual bool ActionEscDispatch(const wchar_t wch,
-                                       const unsigned short cIntermediate,
-                                       const wchar_t wchIntermediate) = 0;
+                                       const std::optional<wchar_t> intermediate) = 0;

```

Comment:
Is it worth making this a `basic_string_view` now, rather than `optional`, so we can support multiple intermediates in the future? The one place I know they're needed is in the SCS sequence for designating character sets (in some cases the second intermediate forms part of the character set identifier).

Technically multiple intermediates are also allowed in CSI sequences, although I don't know of any examples offhand. So `ActionCsiDispatch` is probably still OK as an `optional`, but if you're doing the one, it may be worth doing them both.

Reply:
Hm, I thought about it but I couldn't think of examples. I suppose it doesn't hurt much to pipe it through as an N instead of an optional.

Reply:
This was easy to do, so I just went ahead and did it. Thanks for the push. I like future proofing when it's cheap/easy.